packages/expo-modules-core/src/hooks/useReleasingSharedObject.ts

```diff
@@ -2,7 +2,7 @@
 
 import { DependencyList, useRef, useMemo, useEffect } from 'react';
 
-import type { SharedObject } from '../ts-declarations/SharedObject';
+import { type SharedObject } from '../ts-declarations/SharedObject';
```

Comment:
using `import type {} from 'ts-declarations'` or other type only files make sense to me. that would ensure the file only export types. 

or do you have any particular preference to align our code to use `import { type ...}`?

Reply:
In general, `import type {}` is often perceived as anti-pattern, especially when given file contains mix of imports it often lead to having two, separate imports per file. 

In case of `ts-declarations` which only contains the type, it seems justified, but I'm leaning more into aligning it to just one form, to take out one more decision in the process on the developer side, when adding new code.

I'm also considering adding the following rule to our config, to make sure that we have a consistent pattern in our code base:
* https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/consistent-type-specifier-style.md

The autodetection of types during compile time has been vastly improved in the TypeScript, and hopefully in near-future the `type` imports can be fully dropped, and they will be stripped automatically when compiling JS, see Verbatim Imports flag from TypeScript 5.0:
* https://www.typescriptlang.org/tsconfig/#verbatimModuleSyntax

Reply:
fair enough to me and good to know the eslint rules. thanks for sharing the insight.